script/util/copy-verifier.sh

```diff
@@ -47,73 +45,20 @@ mkdir -p $VERIFIER_FOLDER
 wget -O $VERIFIER_FILE $VERIFIER_SOL_URL
 cp "${GIT_REPO_PATH}/${CODE_DIR_PATH}/Groth16VerifierExtensions.sol" $VERIFIER_EXTENSIONS_FILE
 
-# Use as library instead of contract
-sed -i '' 's/contract Verifier/library Groth16Verifier/' $VERIFIER_FILE
-
-# Use internal instead of public functions
-sed -i '' 's/public view/internal view/' $VERIFIER_FILE
-
-# Read proof argument from memory instead of calldata
-sed -i '' 's/calldata proof/memory proof/' $VERIFIER_FILE
-sed -i '' $'s/calldatacopy(f, proof, 0x100)/mstore(f, mload(add(proof, 0x00)))\
-    mstore(add(f, 0x20), mload(add(proof, 0x20)))\
-    mstore(add(f, 0x40), mload(add(proof, 0x40)))\
-    mstore(add(f, 0x60), mload(add(proof, 0x60)))\
-    mstore(add(f, 0x80), mload(add(proof, 0x80)))\
-    mstore(add(f, 0xa0), mload(add(proof, 0xa0)))\
-    mstore(add(f, 0xc0), mload(add(proof, 0xc0)))\
-    mstore(add(f, 0xe0), mload(add(proof, 0xe0)))/' $VERIFIER_FILE
-
-# Read input argument from memory instead of calldata
-sed -i '' 's/calldata input/memory input/' $VERIFIER_FILE
-sed -i '' 's/calldataload(input)/mload(input)/' $VERIFIER_FILE
-sed -i '' 's/calldataload(add(input, 32))/mload(add(input, 32))/' $VERIFIER_FILE
-sed -i '' 's/calldataload(add(input, 64))/mload(add(input, 64))/' $VERIFIER_FILE
-
-# Import verifier library with renamed filename
-sed -i '' 's/import {Verifier} from ".\/verifier.sol";/import {Groth16Verifier} from ".\/Groth16Verifier.sol";\
-   import {isCDK} from "..\/utils\/Constants.sol";/' $VERIFIER_EXTENSIONS_FILE
+# Rename Query to Groth16VerifierExtension
+sed -i '' 's/contract Query is Verifier {/contract Groth16VerifierExtension is Verifier {/' $VERIFIER_EXTENSIONS_FILE
```

Comment:
Note: the remaining `sed`s will be removed once [this PR](https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/414) is merged and propagates through the system :)

Reply:
sed is removed in this PR. replaced by awk which is more compatible for linux and mac.
https://github.com/Lagrange-Labs/lagrange-lpn-contracts/pull/55